File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int index=-1;
        for(int i=0;i<haystack.size();i++){
            int j=0 , k=i;
            while(haystack[k]==needle[j] && j<needle.size() && k<haystack.size()){
                k++;
                j++;
            }
            if(j==needle.size()){
                index=i;
                break;
            }
        }
        return index;
    }
};

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
        int index = -1;
        for (int i = 0; i < haystack.length(); i++) {
            int j = 0, k = i;
            while (k < haystack.length() && j < needle.length() &&
                   haystack.charAt(k) == needle.charAt(j)) {
                k++;
                j++;
            }

            if (j == needle.length()) {
                index = i;
                break;
            }
        }

        return index;
    }
}